app/_hub/kong-inc/ai-azure-content-safety/how-to/_index.md

```diff
@@ -0,0 +1,201 @@
+all user request content is moderated against a reputable service, to ensure compliance with specific sensitive 
+categories.
+
+This plugin integrates with the [Azure REST API](https://westus.dev.cognitive.microsoft.com/docs/services/content-safety-service-2023-04-30-preview/operations/TextOperations_Analyze) and transmits every user LLM request 
+from users to the Azure Content Safety SaaS *before* proxying to the upstream LLM.
+
+The plugin uses the [**text moderation**](https://learn.microsoft.com/en-us/azure/ai-services/content-safety/quickstart-text?tabs=visual-studio%2Cwindows&pivots=programming-language-rest) operation, and only supports REST API version **2023-10-01**.
+
+To configure the plugin, you set an array of [categories and levels](https://learn.microsoft.com/en-us/azure/ai-services/content-safety/concepts/harm-categories).
+If Azure finds that a piece of content has breached one or more of these levels, 
+the request will be stopped with a 400 status and reported to the Kong log file for auditing.
+
+## Prerequisites
+
+You need an Azure subscription and a Content Safety instance. 
+You can [follow the quickstart from Microsoft](https://learn.microsoft.com/en-us/azure/ai-services/content-safety/quickstart-text?tabs=visual-studio%2Cwindows&pivots=programming-language-rest#prerequisites) 
+to get set up quickly.
+
+Then, as in the [AI Proxy](/hub/kong-inc/ai-proxy/) documentation, create a service, route, and `ai-proxy` plugin
+that will serve as your LLM access point.
+
+## Authentication
+
+In each instance of the plugin, it supports one of:
+
+* Content Safety Key (static key generated from Azure Portal)
+* Managed Identity Authentication
+
+### Content Safety Key Auth
+
+To use a content safety key, you must set the `config.content_safety_key` parameter.
+
+### Managed Identity Auth
+
+To use Managed Identity auth (e.g. Machine Identity on an Azure VM or AKS Pod), you must set `config.use_azure_managed_identity`
+to `true`.
+
+Following this, there are three more parameters that may or may not be required:
+
+* `config.azure_client_id`
+* `config.azure_client_secret`
+* `config.azure_tenant_id`
+
+The client ID is normally required when you want to use a different *user assigned identity* instead of the 
+managed identity assigned to the resource on which Kong is running.
+
+The client secret and tenant ID are usually only used when you are running Kong somewhere outside 
+of Azure, but still want to use Entra ID (ADFS) to authenticate with Content Services.
+
+See the [cloud provider authentication](/hub/kong-inc/ai-proxy/how-to/cloud-provider-authentication/) guide to learn more.
+
+## Examples
+
+Configure the plugin with an array of supported categories as defined in the 
+[Content Services REST API documentation](https://westus.dev.cognitive.microsoft.com/docs/services/content-safety-service-2023-10-01/operations/TextOperations_AnalyzeText), for example using all four
+supported categories in this API version:
```

Comment:
It might actually be more useful to link to https://learn.microsoft.com/en-us/azure/ai-services/content-safety/concepts/harm-categories?tabs=warning (or maybe both?).